src/libraries/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -95,11 +96,21 @@ public override string ReadToEnd()
             return Task.FromResult(ReadLine());
         }
 
+        public override ValueTask<string?> ReadLineAsync(CancellationToken cancellationToken)
```

Comment:
this method should check whether the cancellation was requested and if so, throw